### PR TITLE
[86bxrdq3h][input-number] added Intl.NumberFormat formatter

### PR DESCRIPTION
**Pull request by ilyabrower**
## How has this been tested?
Manually and I've added some tests

## Screenshots (if appropriate):
Screenshot 2024-04-01 at 14 33 01

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [X] Nice improve.

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly or it's not required.
- [X] Unit tests are not broken.
- [X] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [X] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

**Comment by msereniti** [comment]
Looks like you should also add a new documentation example and update component documentation playground 👀

**Comment by ilyabrower** [comment]
> Looks like you should also add a new documentation example and update component documentation playground 👀

No, because we've changed display value for every number inputs

**Comment by msereniti** [comment]
https://github.com/semrush/intergalactic/assets/31261408/4197f0d3-9c6f-43d5-8cf9-3daf024ada00

https://github.com/semrush/intergalactic/assets/31261408/7ca22b54-9f9c-41ed-bf94-92104df343d7

1. When I focus on input it changes it's size. I suppose it should not. (first video)
2. When I'm trying to select last 3 symbols, selection shifts for 1 symbol. (first video)
3. `cmd+z` doesn't work
4. When I'm trying to select symbols in the middle of number, selection collapses (second video)

File: semcore/input/src/Input.jsx [hunk 1]
```diff
@@ -120,15 +120,21 @@ class Input extends Component {
 }
 
 class Value extends Component {
-  static defaultProps = {
-    defaultValue: '',
+  static defaultProps = (props) => {
+    return props.disableUncontrolledValueChange
```
Comment (msereniti) [inline semcore/input/src/Input.jsx:124]: You don't need these changes in this file, isn't it?

Comment (ilyabrower) [reply semcore/input/src/Input.jsx]: yes, removed

File: semcore/input-number/src/InputNumber.jsx [hunk 2]
```diff
@@ -106,51 +180,271 @@ class Value extends Component {
...
+    // Examples:
+    // - Press ArrowLeft: `12,3|4 -> 12|,34`
+    // - Press ArrowRight: `1|,55 -> 1,5|5`
+    const cursorIndex = 2;
+
+    if (event.shiftKey && this.cursorPosition === -1) {
+      this.cursorPosition = element.selectionStart;
+    }
+
+    if (this.cursorPosition !== -1) {
```
Comment (msereniti) [inline semcore/input-number/src/InputNumber.jsx:277]: This tree of ifs looks is not easy to read. Could you somehow refactor it into more compact way?

File: semcore/input-number/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [5.23.0] - 2024-04-02
+
+### Added
+
+- `Intl.NumberFormat` formatter for display value.
```
Comment (msereniti) [inline semcore/input-number/CHANGELOG.md:9]: Let's say something like "Beautiful number formatting ✨ "

Comment (ilyabrower) [reply semcore/input-number/CHANGELOG.md:9]: fixed

File: semcore/input/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [4.23.0] - 2024-04-02
+
+### Changed
+
+- Internal api for uncontrolled props.
```
[the remark this answers is not in the document]
Comment (ilyabrower) [reply semcore/input/CHANGELOG.md:9]: removed

**Review by msereniti — APPROVED** [review]

File: semcore/input-number/src/InputNumber.jsx (hunk 2 above)
Comment (ilyabrower) [reply semcore/input-number/src/InputNumber.jsx]: refactored a little bit

**Pull request merged**